dlrover/python/master/node/dist_job_manager.py

```diff
@@ -790,6 +790,34 @@ def _process_event(self, event: NodeEvent):
             should_relaunch = self._should_relaunch(
                 cur_node, status_change_flow
             )
+
+            # Even _should_relaunch return True by state-machine check,
+            # we need another round of check
+            if should_relaunch:
+                selector = k8s_util.gen_k8s_label_selector_from_dict(
+                    self._get_pod_unique_labels(cur_node)
+                )
+                logger.info(
```

Comment:
This part has actually already been implemented (when generating the event before relaunch), so the issue probably isn't here.

Reply:
event type is MODIFIED and node status is RUNNING from log. so the condition in _process_event is not matched.
we didn't see the log either

Reply:
modified event with deletion timestamp will transfer to deleted event:
https://github.com/intelligent-machine-learning/dlrover/blob/5d070f49da6bde5389b0d282a1712ff6cf27d2b8/dlrover/python/master/watcher/k8s_watcher.py#L232